modules/carto/src/api/maps-api-common.ts

```diff
@@ -77,3 +77,9 @@ type MapInstantiationFormats = Record<
 export function encodeParameter(name: string, value: string | boolean | number): string {
   return `${name}=${encodeURIComponent(value)}`;
 }
+
+export type QueryParameter = {
```

Comment:
This type doesn't support positional arguments in all providers.

[PostgreSQL and Redshift](https://node-postgres.com/features/queries):
```sql
const layer = new CartoLayer({
     data: `select * from users where username=$1`
     queryParameters: ['alasarr']
})
```

[BigQuery positional](https://cloud.google.com/bigquery/docs/parameterized-queries#node.js):
```sql
const layer = new CartoLayer({
     data: `select * from users where username=?`,
     queryParameters: ['alasarr']
})
```

BigQuery named parameters:
```sql
const layer = new CartoLayer({
     data: `select * from users where username=@username`,
     queryParameters: [{ username: 'alasarr' }]
})
```

[Snowflake positional](https://docs.snowflake.com/en/user-guide/nodejs-driver-use.html#binding-statement-parameters) :
```sql
const layer = new CartoLayer({
     data: `select * from users where username=?`,
     queryParameters: ['alasarr' ]
})
```
or
```sql
const layer = new CartoLayer({
     data: `select * from users where username=:1`,
     queryParameters: ['alasarr' ]
})
```

[Databricks ODBC](https://github.com/markdirish/node-odbc#bindparameters-callback)
```sql
const layer = new CartoLayer({
     data: `select * from users where username=?`,
     queryParameters: ['alasarr' ]
})
```

Reply:
hey @alasarr, changes are done as requested since the backend is ready, the documentation updated too!